# CASE: O vendedor de Pirozhki


Vladimir Sidorov desistiu de seu trabalho convencional na KGB e decidiu empreender em 2010 no ramo alimentício. Com suas economias abriu um foodtruck no bairro do Ipiranga – SP especializado na venda de Pirozhki, onde seu grande diferencial era trabalhar com ingredientes importados dando uma qualidade extraordinária ao produto. Após o crescimento das vendas, a modalidade delivery foi implementada para auxiliar a empresa. Vladimir era extremamente popular no bairro, todos o conheciam, mas em 2019 se mudou para um ponto melhor e não tivemos mais notícia. Os últimos dados compartilhados das vendas de Vlad foram disponibilizados antes de sua partida.


In [ ]:

Ano	vendas
2010	723
2011	814
2012	905
2014	1087
2017	1360
2018	1451
2019	1542




1.   Dado o problema acima, qual o melhor algoritmo para descobrir os dados de 2015?
2.   Qual coeficiente angular?
3.   Considerando uma perspectiva estatística, qual o total de venda em 2020 (Apresentar valores)?
4. Considerando uma perspectiva estatística, qual o total de venda em 2021 (Apresentar valores)?
5. Desenvolva um código em python que calcule estatisticamente as vendas em qualquer ano desejado
6. Considerando sua opinião, qual o total de venda em 2021?
7. Considerando sua opinião, qual o total de venda em 2023?



In [ ]:
#1

import pandas as pd
from sklearn.linear_model import LinearRegression

# Dados fornecidos
data = {'Ano': [2010, 2011, 2012, 2014, 2017, 2018, 2019],
        'Vendas': [723, 814, 905, 1087, 1360, 1451, 1542]}

df = pd.DataFrame(data, columns=['Ano', 'Vendas'])


# Crie um modelo de regressão linear
modelo = LinearRegression()

# Ajuste o modelo aos dados
modelo.fit(df[['Ano']], df['Vendas'])

# Faça uma previsão para o ano de 2015
ano_2015 = [[2015]]
previsao_2015 = modelo.predict(ano_2015)

print("Previsão de vendas para 2015:", previsao_2015[0])


Previsão de vendas para 2015: 1178.0


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


In [ ]:
#2
def coeficiente_angular(x1, y1, x2, y2):
    """
    Calcula o coeficiente angular (inclinação) de uma reta que passa pelos pontos (x1, y1) e (x2, y2).

    Args:
        x1 (float): Coordenada x do primeiro ponto.
        y1 (float): Coordenada y do primeiro ponto.
        x2 (float): Coordenada x do segundo ponto.
        y2 (float): Coordenada y do segundo ponto.

    Returns:
        float: Coeficiente angular da reta.
    """
    return (y2 - y1) / (x2 - x1)

# Dados conhecidos
x1 = 2014
y1 = 1087
x2 = 2017
y2 = 1360

# Calcular o coeficiente angular
coef_angular = coeficiente_angular(x1, y1, x2, y2)
print(f"Coeficiente angular da reta: {coef_angular:.2f}")


Coeficiente angular da reta: 91.00


In [ ]:
#3
import numpy as np

from sklearn.linear_model import LinearRegression


# Dados de entrada

anos = np.array([2010, 2011, 2012, 2014, 2017, 2018, 2019]).reshape(-1, 1)

vendas = np.array([723, 814, 905, 1087, 1360, 1451, 1542])



# Criar um modelo de regressão linear

modelo = LinearRegression()



# Ajustar o modelo aos dados

modelo.fit(anos, vendas)



# Fazer previsões

ano_previsto = 2020  # Ano para o qual você deseja fazer uma previsão

vendas_previstas = modelo.predict([[ano_previsto]])



# Imprimir os resultados

print(f"Previsão de vendas para o ano {ano_previsto}: {vendas_previstas[0]:.2f}")

Previsão de vendas para o ano 2020: 1633.00


In [ ]:
#4
import numpy as np

from sklearn.linear_model import LinearRegression


# Dados de entrada

anos = np.array([2010, 2011, 2012, 2014, 2017, 2018, 2019]).reshape(-1, 1)

vendas = np.array([723, 814, 905, 1087, 1360, 1451, 1542])



# Criar um modelo de regressão linear

modelo = LinearRegression()



# Ajustar o modelo aos dados

modelo.fit(anos, vendas)



# Fazer previsões

ano_previsto = 2021  # Ano para o qual você deseja fazer uma previsão

vendas_previstas = modelo.predict([[ano_previsto]])



# Imprimir os resultados

print(f"Previsão de vendas para o ano {ano_previsto}: {vendas_previstas[0]:.2f}")

Previsão de vendas para o ano 2021: 1724.00


In [ ]:
#5
import numpy as np

from sklearn.linear_model import LinearRegression



# Dados de entrada

anos = np.array([2010, 2011, 2012, 2014, 2017, 2018, 2019]).reshape(-1, 1)

vendas = np.array([723, 814, 905, 1087, 1360, 1451, 1542])



# Criar um modelo de regressão linear

modelo = LinearRegression()



# Ajustar o modelo aos dados

modelo.fit(anos, vendas)



# Fazer previsões

ano_previsto = int(input("Digite o ano desejado para a previsão: ")) # Ano para o qual você deseja fazer uma previsão

vendas_previstas = modelo.predict([[ano_previsto]])



# Imprimir os resultados

print(f"Previsão de vendas para o ano {ano_previsto}: {vendas_previstas[0]:.2f}")

Digite o ano desejado para a previsão: 2024
Previsão de vendas para o ano 2024: 1997.00


#6
Baseado no padrão de vendas a tendencia é que no ano de 2021 as vendas cresçam em 5.57% em relação ao ano anterior chegando a 1724 vendas

#7

Baseado no padrão de vendas a tendencia é que no ano de 2023 as vendas cresçam em 5.01% em relação ao ano anterior chegando a 1906 vendas

*   Nome:  Luiza Ignacio Gomes
*   RM: 97797
*   Nome: Raphael Rodrigues Papa
*   RM: 552432

